doc/open-extensions/smart-markers.md

```diff
@@ -0,0 +1,39 @@
+When a client enters a conversation after being offline for a while, such client might want to know what was the last message-id that was marked according to the rules defined in [XEP-0333 - Chat Markers][chat-markers], in order to know where he left of, and build an enhanced UI.
```

Comment:
user might want to know where other participants left off as well, that was the main intention of this module.

also please fix typo - `left off`

Reply:
`where other participants left off` is not what beekeeper needs, so I'm not implementing that part, yet. Would probably require a different IQ, and very likely make it configurable, because some deployments might consider where other users are as private data to those other users only and they shouldn't be fetched otherwise.

Might add a note to state that such functionality will be implemented as well if that makes the doc better 🤔

Reply:
I'm not sure that we want it in open source, if it's too BKPR specific.

The main idea was to let the user fetch all the last chat markers (his own and other participants as well) for a conversation (1-2-1 or a group chat), and I believe that is the protocol that we need in open source.

if you wish to have a common implementation, you may want to add an optional field `[from=<jid>]` to cover BKPR's use case.

Reply:
Putting here a short summary of an offline discussion with @NelsonVides:

- IQ protocol can stay as is.
- a new configuration option for smart markers module can be introduced to cover beekeeper's use case:
   - the behaviour must be consistent for the offline and the online use cases. If a user receives online chat markers, it must be possible to fetch offline chat markers as well.
   - when enabled, this option must ensure filtering of the online chat markers (i.e. no chat markers delivered to the users). As for the carbon copies - the user should receive carbon copies only for his own chat marker messages.
   - when enabled, this option must limit fetching from the DB only to the user's own chat markers. 
   - by default, this option must be disabled to ensure the consistent behaviour with XEP-333.
   - The implementation of this configuration option must be done in a generic way (in mod_smart_markers:get_chat_markers() interface), there should be no difference in code for supporting this option in [mod_offline_chatmarkers.erl](https://github.com/esl/MongooseIM/blob/master/src/offline/mod_offline_chatmarkers.erl) nor in IQ protocol implementation.

Justification: such option allows implementing behaviour similar to a various existing popular messaging application (e.g. slack).